### ISSUE TITLE
HTTP HEAD does not return a response to the client

### ISSUE THREAD

**Issue by erandidesilva** [ballerinalang/ballerina#2486]
Step 1 - Set up a back end service that accepts http head as the http method for the resource
Step 2 - Write a similar code as given below

```
import ballerina.net.http;

@http:BasePath {value:"/http"}
service contentBasedRouting {
    string connection = "http://localhost:5301";

    @http:POST {}
    @http:GET {}
    @http:Path {value:"/callHead"}
    resource httpHeadResource (message m) {
        string resourcePath = "/abc";
        string headerValue;
        message response;
        message request = m;
        http:ClientConnector httpCheck = create http:ClientConnector(connection);
        response = http:ClientConnector.head(httpCheck, resourcePath, request);
        reply response;
    }

}

```

Step 3. Invoke the above service

**Observations**
1. When the communication is analyzed using tcp mon, below was seen that indicates, outgoing and incoming from and to ballerina

**Incoming message from client to ballerina**

```
POST /http/callHead HTTP/1.1
Host: localhost:5302
Connection: keep-alive
Content-Length: 122
Postman-Token: 81fe162a-0a87-173b-6cc4-9334f5faa477
Cache-Control: no-cache
Origin: chrome-extension://fhbjgbiflinjbdggehcddcbncdddomop
User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.86 Safari/537.36
Content-Type: application/json
Accept: */*
Accept-Encoding: gzip, deflate, br
Accept-Language: en-US,en;q=0.8

{
   "Order":{
       "orderId":"ee1a9ec2-c8a5-4afe-8585-74df591f9990",
       "additions":"Chocolate Chip Cookies"
   }
}
```

**Outgoing from ballerina**
```
HEAD /abc HTTP/1.1
Connection: keep-alive
Content-Length: 122
Postman-Token: 0c5da918-1b64-c2bb-f384-b8a420967f5b
Cache-Control: no-cache
Origin: chrome-extension://fhbjgbiflinjbdggehcddcbncdddomop
Content-Type: application/json
Accept: */*
Accept-Encoding: gzip, deflate, br
Accept-Language: en-US,en;q=0.8
HOST: localhost:5301
User-Agent: ballerina/0.88-SNAPSHOT

{
   "Order":{
       "orderId":"ee1a9ec2-c8a5-4afe-8585-74df591f9990",
       "additions":"Chocolate Chip Cookies"
   }
}
```

**Incoming to Ballerina**
```
HTTP/1.1 200 OK
Content-Type: application/json
Content-Encoding: gzip
Content-Length: 145
Server: Jetty(6.1.26)

```
**Outgoing from ballerina to client** 
```
None
```

**Comment by chamil321** [comment]
This issues solved after adding HTTP HEAD annotations to server connector

**Comment by shafreenAnfar** [comment]
As per the above comment. I am closing this issue.

**Comment by shafreenAnfar** [comment]
I am re-opening the issue as we haven't tested with Jetty as in the given sample.

**Comment by chamil321** [comment]
This issue is well fixed with the carbon transport refactor https://github.com/wso2/carbon-transports/pull/345

**Comment by shafreenAnfar** [comment]
As the transport refactor is merged this issue is fixed in 0.93 release.
